packages/iris-grid/src/IrisGridTableModelTemplate.ts

```diff
@@ -379,27 +440,27 @@ class IrisGridTableModel extends IrisGridModel {
         value,
         column.type,
         column.name,
-        formatOverride
+        formatOverride ?? undefined
       );
       this.cacheFormattedValue(x, y, text);
     }
 
     return this.formattedStringData[x][y];
   }
 
-  textForCell(x, y) {
+  textForCell(x: ModelIndex, y: ModelIndex): string | null {
     const text = this.textValueForCell(x, y);
     if (text == null && this.isKeyColumn(x)) {
       const pendingRow = this.pendingRow(y);
-      if (this.pendingDataMap.has(pendingRow)) {
+      if (pendingRow && this.pendingDataMap.has(pendingRow)) {
```

Comment:
`pendingRow != null`
```suggestion
      if (pendingRow != null && this.pendingDataMap.has(pendingRow)) {
```